### ISSUE TITLE
[Bug] Inquiry about Assertion Failure and Logging Issues in AzureCommunicationCalling SDK

### ISSUE THREAD

**Issue by ShawnHuangS** [Azure/azure-sdk-for-ios#1708]
Hello,

I'm encountering an issue with the AzureCommunicationCalling SDK(version 2.6.1) on iOS 15. When frequently toggling the camera, the application crashes with the following assertion failure displayed in the console:

*** Assertion failure in - [SKPPreviewVideoRenderer dealloc](https://github.com/Azure/azure-sdk-for-ios/issues/1625), SKPPreviewVideoRenderer.mm:135
Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Expecting to be on #main thread.'

I have confirmed that I am calling this on the main thread.

To address this issue, I attempted to upgrade to versions 2.7 and 2.8. After upgrading, the crashing problem was resolved. However, both versions seem to generate excessive logs in the console.

Could you please provide guidance on how to handle or reduce the verbosity of logging in the AzureCommunicationCalling SDK (Version 2.7 & 2.8)?

Here is a sample of the logs:
`...............`
`...............`
`...............`
`2024-02-22 16:03:17.343 [#60a7c19c-S] T#32039 S.12 [DEBUG4] [AriaSDK.Storage] 60a7c19c: `
`2024-02-22 16:03:17.343 [#d7884e05-S] T#32039 S.12 [DEBUG4] [AriaSDK.Storage] d7884e05: 2500`
`2024-02-22 16:03:17.344 [#00000000-S] T#6939 [DEBUG1] [azure-communication-services.calling] Telemetry acs_calling_stop_video sent: 1 filtered: 0`
`2024-02-22 16:03:17.344 [#00000000-S] T#6939 [DEBUG1] [azure-communication-services.calling] @000000010f00d200: CallBase::CheckIfOutgoingVideoStreamIsOwned`
`...............`
`...............`
`...............`
`2024-02-22 16:03:17.345 [#00000000-S] T#6939 [DEBUG1] [azure-communication-services.calling] [Telemetry] TelemetryProvider::SubmitTelemetry [priority = 2, name = acs_calling_stop_video]`
`2024-02-22 16:03:17.345 [#ddeae6ef-S] T#6939 [DEBUG4] [AriaSDK] ddeae6ef: 0x109039650 acs_calling_stop_video`
`...............`

Thank you for your assistance.

**Comment by tjprescott** [comment]
@raosanat can you take a look?

**Comment by lucianopa-msft** [comment]
Hi @ShawnHuangS, 

We already fixed the issue with logs, it would be fixed on the next release. Regarding the main assertion failure it is still repro on latest release https://github.com/Azure/Communication/releases/tag/v2.8.0?

**Comment by ShawnHuangS** [comment]
Hi @lucianopa-msft ,

Thank you for your response.

Regarding the assertion failure error, it only occurs in version 2.6 on iOS 15; it doesn't occur on iOS 16 and above. After updating to versions 2.7 and 2.8, the issue also doesn't occur on iOS 15.

To address the log issue, which version should I update to? Is it version 2.8.1 or 2.9? 
Thank you.

**Comment by lucianopa-msft** [comment]
The next release will contain the fix will be 2.9, and probably 2.9.0-beta.1 will have it two once released.

**Comment by ShawnHuangS** [comment]
Hi @lucianopa-msft ,

When I updated to version 2.9.0, the issue of continuous appearance of logs during the call still persists. Could you please let me know if I missed something?


```
．．．．．．
corresponding projected object [Name: IncomingAudioStream, Addr:4444336400] and event handlers.
2024-03-25 10:33:29.442 [#92397648-S] T#89091 [DEBUG4] [auf] 92397648: 29
2024-03-25 10:33:29.965 [#8cc77120-S] T#102667 [DEBUG4] [spl] 8cc77120: success
2024-03-25 10:33:29.965 [#8cc77120-S] T#113211 [DEBUG4] [spl] 8cc77120: success
2024-03-25 10:33:29.965 [#8cc77120-S] T#104971 [DEBUG4] [spl] 8cc77120: success
2024-03-25 10:33:31.549 [#8cc77120-S] T#119299 [DEBUG4] [spl] 8cc77120: success
2024-03-25 10:33:38.065 [#89718566-S] T#93491 S.36 [DEBUG1] [httpstack.rt.Connection] @00000001073b90e0: 89718566: end_of_stream
．．．．．．．
```

Thank you for your assistance.
